docs/cmd/kn_service_create.md

```diff
@@ -39,26 +39,29 @@ kn service create NAME --image IMAGE [flags]
+  -e, --env stringArray            Environment variable to set. NAME=value; you may provide this flag any number of times to set multiple environment variables. To unset, specify the environment variable name followed by a "-" (e.g., NAME-).
+      --env-from stringArray       Config a envfrom with a config map or secret. (config-map | secret):CONFIG_MAP_OR_SECRET_NAME you may provide this flag any number of times. To unset, specify the config map name followed by a "-" (e.g., config-map:CONFIG_MAP_NAME- or secret:SECRET_NAME-).
+      --force                      Create service forcefully, replaces existing service if any.
+  -h, --help                       help for create
+      --image string               Image to run.
+  -l, --label stringArray          Service label to set. name=value; you may provide this flag any number of times to set multiple labels. To unset, specify the label name followed by a "-" (e.g., name-).
+      --limits-cpu string          The limits on the requested CPU (e.g., 1000m).
+      --limits-memory string       The limits on the requested memory (e.g., 1024Mi).
+      --lock-to-digest             keep the running image for the service constant when not explicitly specifying the image. (--no-lock-to-digest pulls the image tag afresh with each new revision) (default true)
+      --max-scale int              Maximal number of replicas.
+      --min-scale int              Minimal number of replicas.
+  -n, --namespace string           List the requested object(s) in given namespace.
+      --no-lock-to-digest          do not keep the running image for the service constant when not explicitly specifying the image. (--no-lock-to-digest pulls the image tag afresh with each new revision)
+  -p, --port int32                 The port where application listens on.
+      --requests-cpu string        The requested CPU (e.g., 250m).
+      --requests-memory string     The requested memory (e.g., 64Mi).
+      --revision-name string       The revision name to set. Must start with the service name and a dash as a prefix. Empty revision name will result in the server generating a name for the revision. Accepts golang templates, allowing {{.Service}} for the service name, {{.Generation}} for the generation, and {{.Random [n]}} for n random consonants. (default "{{.Service}}-{{.Random 5}}-{{.Generation}}")
+      --volume stringArray         Config a volume with a config map or secret. VOLUME_NAME=(config-map|secret):CONFIG_MAP_OR_SECRET_NAME ; you may provide this flag any number of times to define multiple volumes. To unset, specify the volume name followed by a "-" (e.g., /mount/path-).
+      --volume-mount stringArray   Config a volume mount. /mount/path=VOLUME_MOUNT ; you may provide this flag any number of times to mount multiple volumes. To unset, specify the mount path followed by a "-" (e.g., /mount/path-).
```

Comment:
Since the flags defined in `pkg/kn/commands/service/configuration_edit_flags.go` are shared for both of `kn service create` and `kn service update`, this seems inevitable unless the structure is changed. From `--env` or `--label`, the similar description can be found.

Reply:
ok, let's postpone this cleanup for a later release.